### ISSUE TITLE
BigFloat#to_s doesn't show decimal for whole numbers

### ISSUE THREAD

**Issue by watzon** [crystal-lang/crystal#7974]
I could be wrong here, but for sake of consistency I think this makes sense. Here's the scenario:

```crystal
# This works as expected
(1.3).to_big_f.to_s
# => "1.3"

# This however does not
(1.0).to_big_f.to_s
# => "1"
```

As you can see, when the second example is transformed to a String it doesn't show the decimal place. As it is a float I would expect it to always show, at the very least, a zero following the decimal.

**Comment by vlazar** [comment]
Shouldn't these give the same results?

```crystal
"1.1".to_big_f.to_s
# => 1.1

1.1.to_big_f.to_s
# => 1.10000000000000008882
```

**Comment by watzon** [comment]
@vlazar I'd think so

**Comment by Lasvad** [comment]
fix for @watzon initially issue.
I started looking at the problem you identified @vlazar. It looks a bit more complex than the first one. Not sure if it should be a separate issue?

Hopefully I submitted the PR in proper form. Scanned over the contributing guide. Please let me know if something should be added.

Cheers.

**Comment by konovod** [comment]
About second problem: maybe it is not a problem but a correct behavior.
1.1 can't be represented exactly in floating form, only approximately. In a Float32 type it is represented as `1.10000000000000008882`, and in a BigFloat it is represented as `1.10000.(many zeros)...0000012345` (sorry, i don't know exact precision and representation of BigFloat). 

When we convert `1.1` from string to BigFloat we get `1.10000.(many zeros)...0000012345` and it is close enough to 1.1 to be converted to 1.1 on output.

Now when we use 1.1 numeral it has an Float32 type and value of `1.10000000000000008882`, that is close to 1.1 for a Float32, but isvery far from 1.1 by BigFloat standards. So it is no correct to convert it from `1.10000000000000008882` to `1.10000.(many zeros)...0000012345` - we just don't have the information that `1.10000000000000008882` is going to represent `1.1` and not say `1.1000000000000001`.

Possible solution is a BigFloat literals, but that is definitely another issue.

**Comment by asterite** [comment]
Another solution is to convert the float to string and then use that as the initial value for the big float. Because `1.1.to_s` gives "1.1".

**Comment by asterite** [comment]
Closed by #7982
